src/v/serde/parquet/column_writer.cc

```diff
@@ -89,7 +89,7 @@ class buffered_column_writer final : public column_writer::impl {
                   value_memory_usage = sizeof(value_type);
               }
               _current_page_stats.record_value(v);
-              _value_buffer.push_back(std::move(v));
+              _value_buffer.add_value(std::move(v));
           },
```

Comment:
I think `_current_page_memory_usage` can be replaced by the iobuf memory size so we don't have to book keep seperately - does that help?

Reply:
Good point, will have the encoder class self-report its size and remove `_current_page_memory_usage`.